test/integration/http_integration.cc

```diff
@@ -343,11 +343,12 @@ void HttpIntegrationTest::testRouterHeaderOnlyRequestAndResponse(
                                           {":scheme", "http"},
                                           {":authority", "host"},
                                           {"x-lyft-user-id", "123"}};
-  auto response = sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0);
+  auto response = sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0,
+                                                upstream_index);
 
   // The following allows us to test shutting down the server with active connection pool
   // connections.
-  if (!close_upstream) {
+  if (!leave_upstream_open) {
```

Comment:
I'm not sure if this rename is correct (or maybe just needs some comments). Before, we had `close_upstream`, which indicates "the test is going to close upstream, so do/don't do stuff". Now we have `leave_upstream_open`, which indicates...?

Reply:
I think I didn't process that the reset() was on test_server_, not fake_upstreams_. The name "close_upstream" indicated to me that the function would close the upstream, so that's what I assumed was happening.

So, when 'close_upstream' is true, nothing happens, and when it's false, the Envoy under test is shut down, with the upstream told that it's ok to lose the Envoy connected. How about "leave_envoy_running"?

Reply:
I think we should rename to make it clearer who is doing the connection closing. How about rename to `'caller_will_close_upstream`?

Reply:
Hmm, that makes it clearer what it's meant to be used for, but then `caller_will_close_upstream = false` sounds like this function will shut down the upstream.

How about this: maybe the =false case ought to just be its own test function. There are only 2 tests that use it, and they both have absolutely no other special logic in them. I gave splitting testRouterHeaderOnlyRequestAndResponse() a try, and I think it's slightly cleaner now even beyond what we're discussing here; PTAL.

Reply:
Nice, this is much clearer, thanks!